### PR TITLE
fix issue around rotating the encryption key 

### PR DESCRIPTION
**Pull request by jiaqiluo**
Issue: 
- https://github.com/rancher/rancher/issues/36333 
- https://github.com/rancher/rke/issues/2840

Problem:
The bug is found in Rancher.
RKE will fail to rotate the keys in the following scenario:
- provision an RKE cluster with secrets Encryption disabled, which is the default behavior 
- wait for the cluster to be available 
- edit the cluster to set `rotateEncryptionKey: true` and `secretsEncryptionConfig.enable: true`
  
Solutions:
We rotate the encryption key only when updating an existing cluster that has secret Encryption enabled; all other cases will be handled later by reconciling the encryption provider config.  In this particular case, new certs will be generated when enabling the secret encryption.

Tests:
The tests are done in Rancher, I expect to see the same result with standalone RKE.
The steps are exactly the same as the above.
Results: the cluster is updated successfully, rancher logs show the certs are rotated. 

TODO:

- [ ] once the PR is merged, update rancher's dependencies

### CODE REVIEW

File: cmd/up.go
Comment (thedadams) [inline cmd/up.go]: `clusterState.CurrentState.RancherKubernetesEngineConfig.Services.KubeAPI.SecretsEncryptionConfig` is a pointer. Should we have a `nil`-check on that too?

**Review by paynejacob — APPROVED** [review]

File: cmd/up.go
Comment (jiaqiluo) [reply cmd/up.go]: Good catch! 
Fixed.

Comment (jiaqiluo) [reply cmd/up.go]: I will squash commits into one after getting your approval.

**Review by thedadams — DISMISSED** [review]

**Review by thedadams — APPROVED** [review]

**Review by jakefhyde — DISMISSED** [review]

**Pull request merged**
